odxtools/exceptions.py

```diff
@@ -19,5 +20,61 @@ class OdxWarning(Warning):
     """Any warning that happens during interacting with diagnostic objects."""
 
 
-# Mark DecodeError warnings as errors by default
+# Mark DecodeError and EncodeErrpr warnings as errors by default
 warnings.simplefilter("error", DecodeError)
+warnings.simplefilter("error", EncodeError)
+
+#: Specify whether violations of the ODX specification or failed
+#: assumptions in the ODX library code should cause the routine in
+#: question to abort or whether it should continue anyway. Be aware that
+#: in non-strict mode, the behavior of odxtools is undefined and it
+#: might start eating children...
+strict_mode = True
+
+
+def odxraise(message: Optional[str] = None, error_type: Type[Exception] = OdxError) -> NoReturn:
```

Comment:
message should always be included, without a message, it would be very difficult to know what went wrong

Not all applications preserve the call stack.

Reply:
I agree in principle, but that would be *a lot* of work. For now, the primary intention is to be able to look in a debugger what went wrong...